master/buildbot/util/service.py

```diff
@@ -264,6 +268,7 @@ def reconfigServiceWithBuildbotConfig(self, new_config):
                                     **config_sibling._config_kwargs)
 
     def setServiceParent(self, parent):
+        # XXX(sa2ajj): can we make 'master' a simple read-only property instead?
```

Comment:
They are fixed in the current implementation, but I worry that we'll make change to the service hierarchy and go cross-eyed trying to change `self.parent.parent.parent.parent` to `self.parent.parent.parent` in the right subclasses.

You're right - there's not a good case for running multiple masters in the same process in production, but in testing it makes a lot of sense.

I guess I just don't see the issue -- we have `self.master` pretty much everywhere we need it now.  Why change?